### PR TITLE
feat: decrease bundle size, eslint and ts issues to zero 

### PR DESCRIPTION
**Pull request by himanshu-dixit**
# 🔍 Review Summary 
 ### Purpose
Optimize JavaScript bundle size and streamline the release process for the Composio JS SDK.

#### New Feature
- Introduced `bump_version.sh` script to automate version updates.
- Implemented an automated version bump step in the release workflow.

#### Enhancement
- Adjusted Rollup configuration to disable source maps for production, reducing the bundle size.

#### Documentation
- Added `TO_DOs.md` file outlining future tasks.
- Updated setup script to ensure necessary files are included in the distribution package.

#### Chore
- Appointed a new code owner for the JS section.

### Impact
- **Efficiency**: Enhances the release process by automating version bumps.
- **Performance**: Reduces the JavaScript bundle size for production.
- **Planning**: Provides a structured approach for future enhancements with tasks outlined in a dedicated file. 

<details><summary>Original Description</summary>

No existing description found

</details>

----

> [!IMPORTANT]
> Optimizes JS bundle size, automates version updates, and addresses linting issues in the Composio SDK.
> 
>   - **Behavior**:
>     - Introduced `bump_version.sh` script for automated version updates.
>     - Automated version bump in `release_js_sdk.yml` workflow.
>     - Disabled source maps in `rollup.config.mjs` for production to reduce bundle size.
>   - **Documentation**:
>     - Added `TO_DOs.md` for future tasks.
>     - Updated `setup_cli.sh` to include necessary files in distribution.
>   - **Chore**:
>     - Added `@utkarsh-dixit` as a code owner for JS section.
>   - **Linting and TypeScript**:
>     - Changed ESLint command in `lint_js.yml` from `eslint` to `lint`.
>     - Updated ESLint rule in `eslint.config.mjs` to treat `@typescript-eslint/no-explicit-any` as an error.
>     - Added `prettier-plugin-organize-imports` to `package.json` and `.prettierrc`.
>   - **Misc**:
>     - Added `package.dist.json` for distribution configuration.
>     - Removed unused `env` directory and related files.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=ComposioHQ%2Fcomposio&utm_source=github&utm_medium=referral)<sup> for eab7014f55756e90e74b89c491ba2fe8cac31ebb. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: js/rollup.config.mjs [hunk 1]
```diff
@@ -66,11 +66,5 @@ export default [
...
     ]
   },
-  ...generateBundleAndTypeBundles('index'),
-  ...generateBundleAndTypeBundles('frameworks/cloudflare'),
-  ...generateBundleAndTypeBundles('frameworks/langchain'),
-  ...generateBundleAndTypeBundles('frameworks/openai'),
-  ...generateBundleAndTypeBundles('frameworks/langgraph'),
-  ...generateBundleAndTypeBundles('frameworks/vercel'),
-  ...generateBundleAndTypeBundles('sdk/index'),
+  ...generateBundleAndTypeBundles('index')
```
Comment (shreysingla11) [inline js/rollup.config.mjs:69]: The removal of framework bundles (cloudflare, langchain, openai, etc.) is a breaking change that should be documented. Consider adding a migration guide for users who depend on these framework-specific bundles. Also, this change should be reflected in the version bump with a major version increase following semver.

File: js/package.dist.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "composio-core",
+  "version": "0.4.4-beta",
+  "description": "",
+  "main": "index.js",
+  "scripts": {},
```
Comment (shreysingla11) [inline js/package.dist.json:6]: The package.dist.json contains empty objects for 'scripts' and 'lint-staged'. These should either be populated with necessary configurations or removed to keep the distribution package clean and minimal.

**Comment by shreysingla11** [comment]
### Code Review Summary

#### Changes Overview
- Bundle size optimization through removal of framework bundles and sourcemaps
- Introduction of version management system
- Addition of distribution package configuration
- Workflow improvements for release process

#### Concerns and Suggestions
1. **Breaking Changes**:
   - Removal of framework bundles needs proper documentation
   - Version bump should follow semver for breaking changes

2. **Build Configuration**:
   - Consider environment-based sourcemap configuration
   - Package distribution configuration needs cleanup

3. **Version Management**:
   - Add validation for semantic versioning
   - Improve error handling in version bump script

#### Code Quality Rating: 7/10
- **Positives**: Good focus on bundle size optimization, improved release workflow
- **Areas for Improvement**: Breaking changes documentation, error handling, build configuration flexibility

Please address the comments in the review before merging. The changes are generally positive but need some refinement for better maintainability and user experience.

File: js/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "composio-core",
-  "version": "0.4.6",
+  "version": "0.5.0-beta-1",
```
Comment (plxity) [inline js/package.json:3]: Should we define the version format anywhere? I think we have different nomenclature couple of times

Comment (himanshu-dixit) [reply js/package.json:3]: Following, this pattern, can include this in bump_version
- 0.5.0 for normal
- 0.5.0-beta[-(version)[ for rc releases
 
 WDYT?

File: js/package.json [hunk 4]
```diff
@@ -11,11 +11,10 @@
     "openapispec:generate": "npx @hey-api/openapi-ts",
     "run:cli": "ts-node src/cli/index.ts",
     "run:sample": "ts-node sample.ts",
-    "eslint": "eslint 'src/**/*.{ts,js}' --max-warnings=261",
     "prettier": "prettier --write 'src/**/*.{ts,js,cjs}'",
     "prettier:check": "prettier --check 'src/**/*.{ts,js,mjs,cjs}'",
     "build": "rollup -c rollup.config.mjs && ./setup_cli.sh",
-    "lint": "eslint 'src/**/*.ts'",
+    "lint": "eslint 'src/**/*.{ts,js}' --ignore-pattern 'src/sdk/client/**'",
```
[the remark this answers is not in the document]
Comment (himanshu-dixit) [reply js/package.json:17]: True!

By our API generator 3-4 any issues at root level, making it more specific to `src/sdk/client/core/**`. That would be limited to 3-4 core files.
Screenshot 2024-12-18 at 3 37 34 PM

[the remark this answers is not in the document]
Comment (himanshu-dixit) [reply js/package.json:17]: The above should will still check for cli/framework/utils. As it's only ignore `src/sdk/client` path.

File: js/package.json (hunk 3 above)
Comment (plxity) [reply js/package.json:3]: Let's use `rc` keyword, because python SDK uses that. Let's keep it same across both the SDKs.

**Review by plxity — APPROVED** [review]
LGTM!

**Pull request merged**